### PR TITLE
Pause target timer via scripting

### LINKED ISSUE #2504: Add scripts to pause/resume target timer
Basically a script to pause and resume the target timer, useful for cutscenes mid-level or segments not meant to count towards the actual target time, like a little introduction area for example.

### PR DESCRIPTION
**Pull request by tobbi**
Closes #2504

### CODE REVIEW

**Review by mrkubax10 — APPROVED** [review]

File: src/scripting/functions.hpp [hunk 1]
```diff
@@ -95,6 +95,17 @@ void end_cutscene();
  */
 bool check_cutscene();
 
+/**
+ * Pauses the target timer
```
Comment (Vankata453) [inline src/scripting/functions.hpp:99]: Add a full stop.

File: src/scripting/functions.hpp [hunk 2]
```diff
@@ -95,6 +95,17 @@ void end_cutscene();
...
+ */
+void pause_target_timer();
+
+/**
+ * Resumes the target timer
```
Comment (Vankata453) [inline src/scripting/functions.hpp:104]: Also add a full stop.

**Comment by Vankata453** [comment]
Also believe that these functions would be better suited for the Level class in scripting.

**Review by Vankata453 — CHANGES_REQUESTED** [review]
The new functions for the `Level` scripting class have to be added into the `Level` Squirrel class definition in `data/scripts/default.nut`, so they are accessible via `Level.{function}`.

File: src/supertux/game_session.cpp [hunk 3]
```diff
@@ -831,6 +833,24 @@ GameSession::start_sequence(Player* caller, Sequence seq, const SequenceData* da
       lt->stop();
   }
 }
+void 
+GameSession::set_target_timer_paused(bool paused)
+{
+  m_pause_target_timer = paused;
+  
+  for (const auto& obj : m_currentsector->get_objects())
```
Comment (Vankata453) [inline src/supertux/game_session.cpp:841]: `m_currentsector->get_objects_by_type<LevelTime>()`

File: src/supertux/game_session.cpp [hunk 4]
```diff
@@ -831,6 +833,24 @@ GameSession::start_sequence(Player* caller, Sequence seq, const SequenceData* da
...
+  {
+    auto lt = dynamic_cast<LevelTime*>(obj.get());
+    if (lt) {
+      if(paused) {
+        lt->stop();
+      }
+      else {
+        lt->start();
+      }
```
Comment (Vankata453) [inline src/supertux/game_session.cpp:850]: Could be done with no curly brackets.

**Review by Vankata453 — APPROVED** [review]

**Review by mrkubax10 — APPROVED** [review]

**Pull request merged**